rudder-core/src/main/scala/com/normation/rudder/services/policies/write/Cf3PolicyDraft.scala

```diff
@@ -247,3 +203,101 @@ final case class Cf3PolicyDraft(
 
 }
 
+
+/**
+ * A Parameter Entry has a Name and a Value, and can be freely used within the promises
+ * We need the get methods for StringTemplate, since it needs
+ * get methods, and @Bean doesn't seem to do the trick
+ */
+case class ParameterEntry(
+    parameterName : String,
+    parameterValue: String
+) {
+  // returns the name of the parameter
+  def getParameterName() : String = {
+    parameterName
+  }
+
+  // returns the _escaped_ value of the parameter,
+  // compliant with the syntax of CFEngine
+  def getEscapedValue() : String = {
+    ParameterEntry.escapeString(parameterValue)
+  }
+
+  // Returns the unescaped (raw) value of the paramter
+  def getUnescapedValue() : String = {
+    parameterValue
+  }
+}
+
+/*
+ * Escape string to be CFEngine compliant
+ * a \ witll be escaped to \\
+ * a " will be escaped to \"
+ * The parameter may be null (for legacy reason), and it should be checked
+ */
+object ParameterEntry {
+  def escapeString(x: String) : String = {
+    if (x == null)
+      x
+    else
+      x.replaceAll("""\\""", """\\\\""").replaceAll(""""""", """\\"""")
+  }
+}
+
+/**
+ * A container is just a set of key/value parameter and cf3policyDrafts
+ * whose variables have been processed to manage interdependencies
+ * (like unique variable on a node whose value is contributed by
+ * several directives).
+ */
+class Cf3PolicyDraftContainer(
+   val parameters : Set[ParameterEntry],
+   _drafts: Set[Cf3PolicyDraft]
+) extends Loggable {
+
+  val cf3PolicyDrafts = {
+    /*
+     * Here, for each draft, we have to update all previously processed draft.
+     * So the mutmap.
+     * Could have been an recursive algo, to.
+     */
+    val inprocessing = MutMap[Cf3PolicyDraftId, Cf3PolicyDraft]() /* the target policies (the one we wish to have) */
+    for {
+      draft<- _drafts
+    } {
+      val updated = draft.updateAllUniqueVariables(inprocessing.values.toSeq).map(x => (x.id,x))
```

Comment:
I think I had removed/broke them by error in the refactoring, and that some things actually broke following that. But I can't remember what.